runner/master/run.sh

```diff
@@ -27,13 +27,52 @@ export COMPOSERCACHE="${COMPOSERCACHE:-${CACHEDIR}/composer}"
 export CODEDIR="${CODEDIR:-${WORKSPACE}/moodle}"
 export OUTPUTDIR="${WORKSPACE}"/"${BUILD_ID}"
 export ENVIROPATH="${OUTPUTDIR}"/environment.list
+# The PLUGINSTOINSTALL variable could be set to install external plugins in the CODEDIR folder. The following information is needed
+# for each plugin: gitrepo, folder and branch (optional). The plugin fields should be separated by "|" and each plugin should
+# be separated using ";": "gitrepoplugin1|gitfolderplugin1|gitbranchplugin1;gitrepoplugin2|gitfolderplugin2|gitbranchplugin2[...]"
+# Example: "https://github.com/moodlehq/moodle-local_mobile.git|local/mobile|MOODLE_37_STABLE;git@github.com:jleyva/moodle-block_configurablereports.git|blocks/configurable_reports"
+export PLUGINSTOINSTALL="${PLUGINSTOINSTALL:-}"
+
+if [ -n "$PLUGINSTOINSTALL" ];
+then
+  echo ">>> startsection Download external plugins <<<"
+  echo "============================================================================"
+  # Create a temporary folder.
+  PLUGINSTMPDIR=$(mktemp -d -t ci-$(date +%Y-%m-%d-%H-%M-%S)-XXXXXXXXXX)
```

Comment:
Actually, better to do this further up around line 28.
```
export PLUGINDIR="${PLUGINDIR:-${WORKSPACE}/plugins}"
```

And then add this just ebfore the PLUGINSTOINSTALL check:
```
mkdir -p "${PLUGINDIR}"
if [ -n "$PLUGINSTOINSTALL" ];
```

The directory will then always exist. The PLGUINSTOINSTALL just fills it, and after this point we know it always exists.

Reply:
Making this change has the added benefit that for people running regularly, they can have a local copy of the plugins and they will not be cloned each time the script runs.

i.e. they can have their checkouts and just set the env variable to somewhere outside of the workspace.